modules/oe_theme_helper/src/Plugin/Field/FieldFormatter/FeaturedMediaFormatter.php

```diff
@@ -0,0 +1,227 @@
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+      'image_style' => '',
+    ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $element['image_style'] = [
+      '#title' => t('Image style'),
+      '#type' => 'select',
+      '#default_value' => $this->getSetting('image_style'),
+      '#empty_option' => t('None (original image)'),
+      '#options' => image_style_options(FALSE),
+      '#description' => t('Image style to be used if the Media is an image.'),
+    ];
+
+    return $element;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+    $summary = [];
+
+    $image_styles = image_style_options(FALSE);
+    // Unset possible 'No defined styles' option.
+    unset($image_styles['']);
+    // Styles could be lost because of enabled/disabled modules that defines
+    // their styles in code.
+    $image_style_setting = $this->getSetting('image_style');
+    if (isset($image_styles[$image_style_setting])) {
+      $summary[] = t('@image_style image style', ['@image_style' => $image_styles[$image_style_setting]]);
+    }
+    else {
+      $summary[] = t('Original image style.');
+    }
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = [];
+
+    foreach ($items as $delta => $item) {
+      $elements[$delta] = $this->viewElement($item, $langcode);
+    }
+
+    return $elements;
+  }
+
+  /**
+   * Renders a single field item.
+   *
+   * @param \Drupal\Core\Field\FieldItemInterface $item
+   *   The individual field item.
+   * @param string $langcode
+   *   The language code.
+   *
+   * @return array
+   *   The ECL media-container parameters.
+   */
+  protected function viewElement(FieldItemInterface $item, string $langcode): array {
+    $build = ['#theme' => 'oe_theme_helper_featured_media'];
+    $params = ['description' => $item->caption];
+    $media = $item->entity;
+    $cacheability = CacheableMetadata::createFromRenderArray($build);
+
+    if (!$media instanceof MediaInterface) {
+      return [];
+    }
+
+    // Retrieve the correct media translation.
+    $media = $this->entityRepository->getTranslationFromContext($media, $langcode);
+
+    // Caches are handled by the formatter usually. Since we are not rendering
+    // the original render arrays, we need to propagate our caches to the
+    // oe_theme_helper_featured_media template.
+    $cacheability->addCacheableDependency($media);
+
+    // Get the media source.
+    $source = $media->getSource();
+
+    if ($source instanceof OEmbed) {
```

Comment:
We need to expand this so that other available media types are supported: iframes and avportal medias.

Reply:
See oe_theme_preprocess_paragraph__oe_text_feature_media for more info on how to handle this.